osmoutils/ibc.go

```diff
@@ -2,24 +2,29 @@ package osmoutils
 
 import (
 	"encoding/json"
-
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	transfertypes "github.com/cosmos/ibc-go/v4/modules/apps/transfer/types"
 	channeltypes "github.com/cosmos/ibc-go/v4/modules/core/04-channel/types"
 	ibcexported "github.com/cosmos/ibc-go/v4/modules/core/exported"
 )
 
-// NewStringErrorAcknowledgement returns a new instance of Acknowledgement using an Acknowledgement_Error
-// type in the Response field. These errors differ from the IBC errors in that we use custom error strings.
-// NOTE: Acknowledgements are written into state and thus, changes made to error strings included in packet acknowledgements
-// risk an app hash divergence when nodes in a network are running different patch versions of software.
-func NewStringErrorAcknowledgement(err string) channeltypes.Acknowledgement {
-	// ToDo: Do we want to do this or do we want to use the IBC errors and emit the string?
-
-	return channeltypes.Acknowledgement{
-		Response: &channeltypes.Acknowledgement_Error{
-			Error: err,
-		},
+// NewEmitErrorAcknowledgement creates a new error acknowledgement after having emitted an event with the
+// details of the error.
+func NewEmitErrorAcknowledgement(ctx sdk.Context, err error, errorContexts ...string) channeltypes.Acknowledgement {
```

Comment:
Why don't we just move this to ibc-hooks, then we can delete the entire ibc-hooks <> osmoutils dependency

Reply:
sounds good!

Reply:
ahh, no. Just realized it's in osmoutils because rate-limiting is also using it (we could have rate limiting import it from wasm hooks, but it seems dirtier)

Reply:
Didn't we want rate limiting to import ibc-hooks anyway? (This isn't release blocking at all either way though)

Reply:
yeah, that makes sense. We can change it later when rate limits is based off ibc hooks.